app/src/main/java/com/appunite/loudius/network/utils/ApiCallUtil.kt

```diff
@@ -38,6 +38,9 @@ suspend fun <T> safeApiCall(
 }
 
 private fun getApiErrorMessageIfExist(throwable: HttpException) = try {
+    // suggestion: it would be better to re-use the same gson, instead of creating new one here with the default settings.
+    // suggestion: instead of using string(), it's optimal to use charStream
+    // bug: if error body is null, we shouldn't crash the app.
     val errorResponse = Gson().fromJson(
         throwable.response()?.errorBody()?.string(),
         DefaultErrorResponse::class.java,
```

Comment:
1/2. Agree with that. 👍 
3. There is no crushing the app on null currently. We are catching the exception and returning that the error message is null. I don't see the bug here.